emscripten-simulated-jerry-api/jerry.js

```diff
@@ -148,6 +151,81 @@
       this._jerryToHostValueMap = {};
       this._nextObjectRef = 1;
       this._jerryInternalPropsWeakMap = new WeakMap();
+
+      if (!Module.Promise) {
+        console.warn('Please set Module.Promise to a promise implementation to use for JerryScript')
+      } else {
+        // Patch up the Promise.prototype then/catch implementations.
+        // The patched versions should create a function that when called, pushes the original callback arg to the jobQueue
+        var _then = Module.Promise.prototype.then;
+        var _catch = Module.Promise.prototype.catch;
+        var _identify = function(x) {return x};
+        var _thrower = function(x) {throw x};
+        var jobQueue = this.jobQueue;
+
+        // Modify the `then`.
+        Module.Promise.prototype.then = function (cb1, cb2) {
+          var res, rej;
+          var ret = new Module.Promise(function(f, r) {
+            res = f;
+            rej = r;
+          });
+          if (typeof cb1 !== 'function') {
+            cb1 = _identify;
+          }
+          if (typeof cb2 !== 'function') {
+            cb2 = _thrower;
```

Comment:
Should be `newCb1` and  `newCb2` here?

Reply:
No, should be cb1/cb2, the newCb's duty is push a job (whose job is calling the cb1/cb2) into the jobqueue.

Reply:
Oh I just realize that it's OK to use newCb, and because we dont need to handle the `_identify` and `_thrower` in our jobqueue, we just let the host promise jobqueue to handle it.
What's more, we dont even provide the  the impl of _identity and _thrower, we can just keep the original cb, because the origianl `then` impl will convert them into identity and thrower